### DISCUSSION TITLE
Avaliable to clone labels between repositories

### DISCUSSION THREAD

**Discussion by MoYuan-CN** [community/community#122983] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

Configuration same issues labels in two(or more) repos are troublesome and boring.
Is that avaliable to add a new feature call "Label Clone" to let user can clone labels into a new repo that they have permission?

**Comment by mickeygousset** [comment, marked as answer]
**For new repos:**

Create a repository template that already has all your labels defined. Then, when you create a new repo using that template, the labels will be there.

**For existing repos:**

If you wanted to replicate the labels from on repository to another, you could use the rest apis. Do something like this (NOTE: I have not run this code, thanks GitHub Copilot!):

Get the labels from the source repo:

`curl -s -H "Authorization: token your_token" https://api.github.com/repos/source_owner/source_repo/labels > labels.json`

Run this shell script to take the labels from the source and create them in the target:

```
for row in $(cat labels.json | jq -r '.[] | @base64'); do
    name=$(echo ${row} | base64 --decode | jq -r '.name')
    color=$(echo ${row} | base64 --decode | jq -r '.color')
    description=$(echo ${row} | base64 --decode | jq -r '.description')

    curl -X POST -H "Authorization: token your_token" \
        -d "{\"name\":\"$name\", \"color\":\"$color\", \"description\":\"$description\"}" \
        https://api.github.com/repos/target_owner/target_repo/labels
done

```

If you prefer, you could also use the GitHub CLI, it makes it a little cleaner:

`gh api repos/source_owner/source_repo/labels > labels.json`

```

for row in $(cat labels.json | jq -r '.[] | @base64'); do
    name=$(echo ${row} | base64 --decode | jq -r '.name')
    color=$(echo ${row} | base64 --decode | jq -r '.color')
    description=$(echo ${row} | base64 --decode | jq -r '.description')

    gh api -X POST /repos/target_owner/target_repo/labels -f name="$name" -f color="$color" -f description="$description"
done

```

**Reply by MoYuan-CN** [reply]
Which token permission(s) is required?

**Reply by mickeygousset** [reply]
repo scope should work

**Reply by MoYuan-CN** [reply]
I try it, and work successfully.
Thanks for help.
